File: src/main/java/by/matrosov/codeforces/globalround3/Task2.java
package by.matrosov.codeforces.globalround3;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Task2 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        String[] firstLine = reader.readLine().split(" ");
        String[] secondLine = reader.readLine().split(" ");
        String[] thirdLine = reader.readLine().split(" ");

        int n = Integer.parseInt(firstLine[0]);
        int m = Integer.parseInt(firstLine[1]);
        int ta = Integer.parseInt(firstLine[2]);
        int tb = Integer.parseInt(firstLine[3]);
        int k = Integer.parseInt(firstLine[4]);

        int[] a = new int[n];
        int[] b = new int[m];

        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(secondLine[i]) + ta;
        }

        for (int i = 0; i < m; i++) {
            b[i] = Integer.parseInt(thirdLine[i]);
        }

        int result = -1;
        if (k < Math.min(n,m)){

            int j = 0;
            for (int i = 0; i <= k; i++) {

                while (j < m && b[j] < a[i]){
                    j++;
                }

                if (j + k - i < m){
                    result = Math.max(result, b[j+k-i] + tb);
                }else {
                    result = -1;
                    break;
                }
            }
        }
        System.out.println(result);
    }

}

File: src/main/kotlin/by/matrosov/codeforces/kotlin_heroes/Task2.kt
package by.matrosov.codeforces.kotlin_heroes

import kotlin.math.max

fun main() {
    readLine()!!.toInt()

    val list = readLine()!!.split(" ").map { it.toInt() }

    var firstMax = 0
    var secondMax = 0
    var maxOfTwo: Int
    var count = 0

    for (i in list.indices){
        if (firstMax <= list[i] && secondMax <= list[i]){
            maxOfTwo = max(firstMax, secondMax)
            firstMax = list[i]
            secondMax = maxOfTwo
        }else if (firstMax <= list[i]){
            firstMax = list[i]
        }else if (secondMax <= list[i]){
            secondMax = list[i]
        }else{
            count++
        }
    }
    println(count)
}